app/test/testMetadata.js

```diff
@@ -178,4 +176,50 @@ describe('Metadata API Tests', function() {
         done();
       });
   });
+
+  it('should only return absolute URLs', (done) => {
+    const baseUrl = 'http://www.example.com';
+    const relativeImageUrl = '/media/image.png';
+    const absoluteImageUrl = baseUrl + relativeImageUrl;
+    const relativeIconUrl = '/rich-icon.png';
+    const absoluteIconUrl = baseUrl + relativeIconUrl;
+
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${relativeImageUrl}" />
+            <link rel="icon" href="${relativeIconUrl}" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+
+    const expectedMetadata = JSON.parse(JSON.stringify(exampleMetadata));
```

Comment:
Not sure if you want to create some helper function now that we're doing this object deep-clone logic in 2+ places.